npe2/manifest/contributions/_keybindings.py

```diff
@@ -1,22 +1,22 @@
 from typing import Optional
 
-from pydantic import BaseModel
 from pydantic.fields import Field
 
 from ..utils import Executable
 
 
-# the existence of the command is not validated at registration-time,
-# but rather at call time... (since commands from other extensions can be called)
-class KeyBindingContribution(BaseModel, Executable):
+# TODO: uncomment code when corresponding features enabled in napari
+class KeyBindingContribution(Executable):
     command: str = Field(
         description="Identifier of the command to run when keybinding is triggered."
     )
+    # the existence of the command is not validated at registration-time,
+    # but rather at call time... (since commands from other extensions can be called)
     key: str = Field(
         description="Key or key sequence (separate keys with plus-sign and sequences "
-        "with space, e.g. Ctrl+O and Ctrl+L L for a chord)."
+        "with space, e.g. Ctrl+O"  # and Ctrl+L L for a chord)."
     )
     mac: Optional[str] = Field(description="Mac specific key or key sequence.")
     linux: Optional[str] = Field(description="Linux specific key or key sequence.")
     win: Optional[str] = Field(description="Windows specific key or key sequence.")
-    when: Optional[str] = Field(description="Condition when the key is active.")
+    # when: Optional[str] = Field(description="Condition when the key is active.")
```

Comment:
Maybe move the comment on line 8 down here since this is the only commented out feature.

There are when clauses on other contributions (e.g. menus). Is it that when-clauses are not enabled in general, or just for key-bindings?

I'm kind of tempted to suggest leaving the field uncommented and just make it non-doc. Not sure what the right approach is here.

Reply:
What's required on the napari side to enable this feature @nclack?

Reply:
I'm not quite sure of the status, but there are `ContextKey` objects from `app_model` scattered around the napari code base so I suspect there's some support already.